services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/oauth/OpenIDConnectManagerImplUnitTest.java

```diff
@@ -0,0 +1,715 @@
+		
+		// now let's doctor the authorization time stamp:
+		String code = authResponse.getAccess_code();
+		String decrypted = mockStackEncrypter.decryptStackEncryptedAndBase64EncodedString(code);
+		
+		OIDCAuthorizationRequest authRequestFromCode = new OIDCAuthorizationRequest();
+		JSONObjectAdapter adapter = new JSONObjectAdapterImpl(decrypted);
+		authRequestFromCode.initializeFromJSONObject(adapter);
+		// authorized 100 sec ago (expiration is 60 sec)
+		authRequestFromCode.setAuthorizedAt(new Date(authRequestFromCode.getAuthorizedAt().getTime()-100000L));
+		// now turn back into an authorization code
+		try {
+			authRequestFromCode.writeToJSONObject(adapter);
+		} catch (JSONObjectAdapterException e) {
+			throw new RuntimeException(e);
+		}
+		String serializedAuthorizationRequest = adapter.toJSONString();
+		String modifiedCode = mockStackEncrypter.encryptAndBase64EncodeStringWithStackKey(serializedAuthorizationRequest);
+
+		// method under test
+		try {
+			openIDConnectManagerImpl.getAccessToken(modifiedCode, OAUTH_CLIENT_ID, REDIRCT_URIS.get(0), OAUTH_ENDPOINT);
+			fail("IllegalArgumentException expected");
+		}  catch (IllegalArgumentException e) {
+			// as expected
+		}
+
+	}
+
+	@Test
+	public void testGetAccessToken_mismatchRedirectURI() {
+		OIDCAuthorizationRequest authorizationRequest = createAuthorizationRequest();
+
+		OAuthAuthorizationResponse authResponse = openIDConnectManagerImpl.authorizeClient(userInfo, authorizationRequest);
+		// method under test
+		try {
+			openIDConnectManagerImpl.getAccessToken(authResponse.getAccess_code(), OAUTH_CLIENT_ID, "wrong redirect uri", OAUTH_ENDPOINT);
+			fail("IllegalArgumentException expected");
+		}  catch (IllegalArgumentException e) {
+			// as expected
+		}
+
+	}
+	
+	@Test
+	public void testGetAccessToken_noOpenIdScope() {
+		OIDCAuthorizationRequest authorizationRequest = createAuthorizationRequest();
+		authorizationRequest.setScope(null); // omit the 'openid' scope.  This should suppress the idToken
+
+		OAuthAuthorizationResponse authResponse = openIDConnectManagerImpl.authorizeClient(userInfo, authorizationRequest);
+		String code = authResponse.getAccess_code();
+
+		String expectedIdToken = "ID-TOKEN";
+		when(oidcTokenHelper.createOIDCIdToken(eq(OAUTH_ENDPOINT), anyString(), eq(OAUTH_CLIENT_ID), anyLong(), 
+				eq(NONCE), eq(now.getTime()/1000L), anyString(), userInfoCaptor.capture())).thenReturn(expectedIdToken);
+		
+		String expectedAccessToken = "ACCESS-TOKEN";
+		when(oidcTokenHelper.createOIDCaccessToken(eq(OAUTH_ENDPOINT), anyString(), eq(OAUTH_CLIENT_ID), anyLong(),
+				eq(now.getTime()/1000L), anyString(), scopesCaptor.capture(), claimsCaptor.capture())).thenReturn(expectedAccessToken);
+		
+		// method under test
+		OIDCTokenResponse tokenResponse = openIDConnectManagerImpl.getAccessToken(code, OAUTH_CLIENT_ID, REDIRCT_URIS.get(0), OAUTH_ENDPOINT);
+		
+		assertNull(tokenResponse.getId_token());
+
+		assertNotNull(tokenResponse.getAccess_token());
+	}
```

Comment:
verify(oidcTokenHelper, never()). createOIDCIdToken(....); (and then don't need the mock